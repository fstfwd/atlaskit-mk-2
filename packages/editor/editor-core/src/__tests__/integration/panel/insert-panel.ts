import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';

BrowserTestCase(
  'Inserts a panel on fullpage',
  { skip: ['edge', 'ie', 'safari'] },
  async client => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);

    // await browser.waitForSelector(editable);
    // await browser.click(editable);
    await browser.waitForSelector(fullpage.placeholder);
    await browser.click(fullpage.placeholder);

    await quickInsert(browser, 'Panel');
    // type some text

    await browser.type(editable, 'this text should be in the panel');

    // expect(await browser.isVisible('.media-single')).toBe(true);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
