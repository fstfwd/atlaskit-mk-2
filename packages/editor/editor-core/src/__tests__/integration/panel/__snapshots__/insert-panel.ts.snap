// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inserts a panel on fullpage 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "hello have a link ",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
